get-public-key/Main.hs

```diff
@@ -0,0 +1,21 @@
+-- | Extract a public key from Ristretto-flavored PrivacyPass signing key read from stdin.
+module Main
+  ( main
+  ) where
+
+import Prelude hiding
+  ( putStrLn
+  , getLine
+  )
+
+import Data.Text.IO
+  ( putStrLn
+  , getLine
+  )
```

Comment:
Do you need both of these imports?

Reply:
More or less.  `getLine` to read the private key from stdin as a `Data.Text.Text` and then `putStrLn` to write the `Data.Text.Text` that represents the public key to stdout.

There's probably an entirely different way to manage it but as far as I can tell sticking entirely to `Data.Text.IO` functions is reasonable.

Reply:
Do you also need to import the version from `Prelude`?

Reply:
```
import Prelude hiding
  ( putStrLn
  , getLine
  )
```

has the opposite meaning.  It means "add everything from Prelude to this namespace (this is the default if you don't do anything) except putStrLn and getLine".  So this is (more or less) necessary to avoid having the next import make these two names ambiguous.

The `Prelude` import could be dropped but then these two names have to be referred to qualified (eg `Data.Text.IO.getLine`) to avoid the ambiguity.